builder/Dockerfile

```diff
@@ -22,13 +22,15 @@ RUN apt-get update && \
 RUN wget --quiet https://developer.arm.com/-/media/Files/downloads/gnu/14.2.rel1/binrel/arm-gnu-toolchain-14.2.rel1-x86_64-arm-none-eabi.tar.xz -O - \
         | tar -xJ -C /opt
 
-RUN python3 -m pip install -U pip setuptools \
-    && python3 -m pip install \
+# Python 3.11 introduced the managed environment breakage aka PEP 668.
```

Comment:
I don't disagree with this at all (I suggested exactly this myself in the MSYS build scripts), but I have since come across [`uv`](https://docs.astral.sh/uv/) thanks to Raphael, and it makes package management and venv effortless... . Worth considering at a future time, if not now.

Reply:
Your solution is implemented in multiple files. If not too challenging I vote for changing now,

Reply:
Okay so I've had a play with uv and replaced you magic with this hack
```
RUN wget -qO- https://astral.sh/uv/install.sh | sh && \
    . $HOME/.local/bin/env && \
    uv venv && \
    uv pip install -U pip setuptools && \
    uv pip install \
    asciitree \
    jinja2 \
    pillow \
    aqtinstall \
    lz4 \
    pyelftools

RUN . $HOME/.local/bin/env && \
    uv run aqt install-qt --outputdir ${QT_INSTALL_DIR} ${QT_HOST} ${QT_TARGET} ${QT_VERS_STR} ${QT_ARCH} ${QT_MODULES}
```
It works but does not save any coding in fact means every python script run needs to be amended. However I stand to be duly corrected on matters such as these.